src/main/java/com/nutomic/syncthingandroid/activities/LogActivity.java

```diff
@@ -0,0 +1,168 @@
+package com.nutomic.syncthingandroid.activities;
+
+import android.content.Intent;
+import android.os.AsyncTask;
+import android.os.Bundle;
+import android.support.v7.widget.ShareActionProvider;
+import android.util.Log;
+import android.view.Menu;
+import android.view.MenuInflater;
+import android.view.MenuItem;
+import android.widget.EditText;
+
+import android.support.v4.view.MenuItemCompat;
+import android.widget.ScrollView;
+import android.widget.TextView;
+
+import com.nutomic.syncthingandroid.R;
+
+import java.io.BufferedReader;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+/**
+ * Shows the log information from Syncthing.
+ */
+public class LogActivity extends SyncthingActivity {
+
+    private final static String TAG = "LogActivity";
+
+    private TextView mLog;
+    private boolean mSyncthingLog = true;
+    private AsyncTask mFetchLogTask;
+    private ScrollView mScrollView;
+    private Intent mShareIntent;
+
+    /**
+     * Initialize Log.
+     */
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        setContentView(R.layout.log_activity);
+        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+
+        if (savedInstanceState != null) {
+            mSyncthingLog = savedInstanceState.getBoolean("syncthingLog");
+        }
+
+        mLog = (TextView) findViewById(R.id.log);
+        mScrollView = (ScrollView) findViewById(R.id.scroller);
+
+        updateLog();
+    }
+
+    @Override
+    protected void onSaveInstanceState(Bundle outState) {
+        super.onSaveInstanceState(outState);
+        outState.putBoolean("syncthingLog", mSyncthingLog);
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        MenuInflater inflater = getMenuInflater();
+        inflater.inflate(R.menu.log_list, menu);
+
+        // Add the share button
+        MenuItem shareItem = menu.findItem(R.id.menu_share);
+        if (mSyncthingLog) {
+            shareItem.setTitle(R.string.log_android_title);
+        } else {
+            shareItem.setTitle(R.string.log_syncthing_title);
+        }
+        ShareActionProvider actionProvider = (ShareActionProvider) MenuItemCompat.getActionProvider(shareItem);
+        mShareIntent = new Intent();
+        mShareIntent.setAction(Intent.ACTION_SEND);
+        mShareIntent.setType("text/plain");
+        mShareIntent.putExtra(android.content.Intent.EXTRA_TEXT, mLog.getText());
+        actionProvider.setShareIntent(mShareIntent);
+
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case R.id.switch_logs:
+                mSyncthingLog = !mSyncthingLog;
+                if (mSyncthingLog) {
+                    item.setTitle(R.string.log_android_title);
+                } else {
+                    item.setTitle(R.string.log_syncthing_title);
```

Comment:
Is this a no go without ternary? Golang devs decided not to support ternary operators because they (easily) introduce unreadable/broken code. After spending some time in Go, I actually like their reasoning, keep it super simple and stupid, without fancyness